### DISCUSSION TITLE
setup on windows without visual studio

### DISCUSSION THREAD

**Discussion by RemiBraizet** [microsoft/vcpkg#41913] [Q&A] [answered]
Hi, 

I have msys2 / mingw64 on which there is gcc, GNU make and g++
I do not use visual studio 
I try to cross compile using the same Makefile, either via **WSL** or **powershell** ( and have the possibility to either **dynlink** or **staticlink** )
 
it works using wsl, vcpkg is properly setup
but when i run the command for powershell:

```
.\dependencies\vcpkg_installation\vcpkg.exe install --x-manifest-root=dependencies/ --x-install-root=dependencies/ --triplet=x64-mingw-dynamic
error: in triplet x64-windows: Unable to find a valid Visual Studio instance
Could not locate a complete Visual Studio instance
```

- i've seen that you can install vcpkg directly in mingw64 but i'd like to avoid it since i already have it in my project folder 
- Im not sure if i should be creating a custom triplet

did i miss a step ?

**Comment by dg0yt** [comment]
ATM you are building for mingw on a msvc host.  
You want to change the host triplet to a mingw one.

**Reply by RemiBraizet** [reply]
Like this `--triplet=x64-mingw-static --host-triplet=x64-mingw-static`
or `--triplet=x64-windows --host-triplet=x64-mingw-static` ?
Ive tried both in powershell and WSL 
here a gist with some of the error ouput:
https://gist.github.com/RemiBraizet/e52a4a996858f7a5385b8ef0ec813b59

**Reply by dg0yt** [reply]
Overriding the host triplet also changes the default target triplet, so `--host-triplet=x64-mingw-static` is enough as a starting point.

`--triplet=x64-windows` would select an MSVC target.

> here a gist with some of the error ouput:
> <script src="https://gist.github.com/RemiBraizet/e52a4a996858f7a5385b8ef0ec813b59.js"></script>

Unreadable format.

**Reply by dg0yt** [reply]
FTR you can also use environment variables, so you don't have to repeat the arguments every time.  I source a single PS script to initialize `PATH` and triplet.

**Reply by RemiBraizet** [reply]
i've updated the link 
I runned as you said with the host triplet but still encounter the error

**Reply by dg0yt** [reply, marked as answer]
The log now shows paths like this
~~~
\\wsl.localhost\Ubuntu\home\...
~~~
Try mapping the path to a drive letter.

And don't try to share the same vcpkg dir between Windows and Ubuntu without explicitly using distinct install dirs. They will not agree on the host triplet, and of course it breaks vcpkg's ABI tracking.

**Reply by RemiBraizet** [reply]
Okay thanks i got it running now
